src/Engines/QueryBuilderEngine.php

```diff
@@ -102,20 +102,33 @@ public function filtering()
     {
         $this->query->where(
             function ($query) {
-                $keyword = $this->setupKeyword($this->request->keyword());
+                $globalKeyword = $this->setupKeyword($this->request->keyword());
+                $queryBuilder = $this->getQueryBuilder($query);
+
                 foreach ($this->request->searchableColumnIndex() as $index) {
                     $columnName = $this->getColumnName($index);
-
+                    // check if custom column filtering is applied
                     if (isset($this->columnDef['filter'][$columnName])) {
-                        $method     = Helper::getOrMethod($this->columnDef['filter'][$columnName]['method']);
-                        $parameters = $this->columnDef['filter'][$columnName]['parameters'];
-                        $this->compileColumnQuery(
-                            $this->getQueryBuilder($query),
-                            $method,
-                            $parameters,
-                            $columnName,
-                            $keyword
-                        );
+                        $columnDef = $this->columnDef['filter'][$columnName];
+                        // check if global search should be applied for the specific column
+                        $applyGlobalSearch = count($columnDef['parameters']) == 0 || end($columnDef['parameters']) !== false;
```

Comment:
No need to filter out global search when we use `$queryBuilder->addNestedWhereQuery($whereQuery, 'or');`. This condition can be removed now.